Use of auto in lambda.

`gcc` complains about missing standard declaration, when using `auto` in lambda expressions.
```c
/build/alice-vision-git/src/alice-vision-git/src/aliceVision/system/gpu.hpp:67:54: error: use of ‘auto’ in lambda parameter declaration only available with -std=c++14 or -std=gnu++14
   std::sort(devices.begin(), devices.end(), [](const auto& a, const auto& b) {
```
Os: Linux - Arch, gcc-8.2.1

--- a/src/aliceVision/system/gpu.hpp
+++ b/src/aliceVision/system/gpu.hpp
@@ -63,8 +63,9 @@ inline std::vector<std::pair<cudaDeviceProp, int>> getSortedGpuDevices(int minCo
   int minComputeCapabilityMinor,
   int minTotalDeviceMemory = 0)
 {
-  std::vector<std::pair<cudaDeviceProp, int>> devices = getValidGpuDevices(minComputeCapabilityMajor, minComputeCapabilityMinor, minTotalDeviceMemory);
-  std::sort(devices.begin(), devices.end(), [](const auto& a, const auto& b) {
+  using DeviceAndIndex = std::pair<cudaDeviceProp, int>;
+  std::vector<DeviceAndIndex> devices = getValidGpuDevices(minComputeCapabilityMajor, minComputeCapabilityMinor, minTotalDeviceMemory);
+  std::sort(devices.begin(), devices.end(), [](const DeviceAndIndex& a, const DeviceAndIndex& b) {
     return a.first.multiProcessorCount > b.first.multiProcessorCount;
   });
   return devices;

